Gravity/Gravity.Test.Integration/RSAPI_IntegrationTest.cs

```diff
@@ -0,0 +1,395 @@
+            }
+            catch (Exception ex)
+            {
+                throw new Exception("Error encountered in " + System.Reflection.MethodBase.GetCurrentMethod().Name, ex);
+            }
+            finally
+            {
+                Console.WriteLine("Ending Test case " + System.Reflection.MethodBase.GetCurrentMethod().Name);
+            }
+        }
+
+        [TestCase("LongTextField", TestValues.LongTextFieldValue)]
+        [TestCase("FixedTextField", TestValues.String100Length)]
+        [TestCase("IntegerField", -1)]
+        [TestCase("BoolField", true)]
+        [TestCase("DecimalField", 123.45)]
+        [TestCase("CurrencyField", 5648.54)]
+        [Test, Description("Verify Object Long Text Field Created")]
+        //need object fields, could get a little more difficult
+        public void Valid_Gravity_Object_Create_Field_Type<T>(string objectPropertyName, T sampleData)
+        {
+            Console.WriteLine(System.Reflection.MethodBase.GetCurrentMethod().Name + " Created");
+
+            try
+            {
+                //Arrange
+                Console.WriteLine("Starting Arrangement for property...." + objectPropertyName);
+
+                GravityLevelOne testObject = new GravityLevelOne();
+                testObject.Name = "TestObjectCreate_" + objectPropertyName + Guid.NewGuid().ToString();
+
+                Guid testFieldGuid = BaseDto.GetCustomAttributeOfProperty<GravityLevelOne, Guid>(objectPropertyName, x => x.FieldGuid);
+                //can get rid of cast once FieldType is created as RdoFieldType and not int
+                RdoFieldType fieldType = (RdoFieldType)BaseDto.GetCustomAttributeOfProperty<GravityLevelOne, int>(objectPropertyName, x => x.FieldType);
+
+                //need this mess because when passing in tests for decimal and currency System wants to use double and causes problems
+                switch (fieldType)
+                {
+                    case RdoFieldType.Currency:
+                    case RdoFieldType.Decimal:
+                        testObject.SetValueByPropertyName(objectPropertyName, Convert.ToDecimal(sampleData));
+                        break;
+                    default:
+                        testObject.SetValueByPropertyName(objectPropertyName, sampleData);
+                        break;
+                }
+
+                _client.APIOptions.WorkspaceID = _workspaceId;
+
+                Console.WriteLine("Arrangement Complete....");
+
+                //Act
+                Console.WriteLine("Starting Act....");
+
+                var newRdoArtifactId = _testObjectHelper.CreateTestObjectWithGravity(testObject);
+
+                //read artifactID from RSAPI
+                RDO newObject = _client.Repositories.RDO.ReadSingle(newRdoArtifactId);
+
+                FieldValue field = newObject.Fields.Get(testFieldGuid);
+
+                object newObjectValue = null;
+
+                switch (fieldType)
```

Comment:
just use `field.Value` and remove the switch. The equality statment will still work, as these `mValueAs` properties are all just casts.